python/Application_Companion/memory_usage.py

```
@@ -0,0 +1,122 @@
+        self._log_settings = log_settings
+        self._configurations_manager = configurations_manager
+        self.__logger = self._configurations_manager.load_log_configurations(
+                                        name=__name__,
+                                        log_configurations=self._log_settings)
+        self.__logger.debug("logger is configured.")
+        if not path_to_read_stats:
+            self._path_to_read_stats = path_to_read_stats
+        else:
+            # NOTE: The /proc/[pid]/smaps file is present only if the
+            # CONFIG_PROC_PAGE_MONITOR kernel configuration option is enabled.
+            self._path_to_read_stats = os.path.join(
+                '/proc/', str(process_id), 'smaps')
+
+        # setup memory utilization metrics
+        self.__memory_metrics = ['Vss', 'Rss', 'Pss',
+                                 'Private_Clean', 'Private_Dirty']
+        self.__memory_usage = {}
+
+    def __generate_line_that_contains(self, __memory_metrics, fp):
+        # read line by line in file
+        for line in fp:
+            # check if line contains any of the metrics
+            for metric in __memory_metrics:
+                if metric in line:
+                    # return line
+                    yield line
+
+    # TODO: move general purpose helper functions to util
+    def __split_values(self, line, delimiter):
+        return (map(str.strip, line.split(delimiter)))
+
+    def __get_current_memory_usage(self, file):
+        with open(file, "r") as fp:
+            for line in self.__generate_line_that_contains(
+                                    self.__memory_metrics, fp):
+                # split into key, value
+                key, memory_kb = self.__split_values(line, ':')
+                # strip off "kB"
+                memory, _ = self.__split_values(memory_kb, ' ')
+                self.__memory_usage.setdefault(key, []).append(
+                    float(memory)/1024)  # convert to MiB
```

If memory can not be converted to a float or is zero this line might fail hard.
Maybe put a try around this?

---

yes, handled this case in new commit